app/models/journal_entry.rb

```diff
@@ -282,7 +282,8 @@ def self.states
     errors.add(:items, :empty) unless items.any?
     errors.add(:balance, :unbalanced) unless balance.zero?
     errors.add(:real_balance, :unbalanced) unless real_balance.zero?
-    errors.add(:printed_on, :not_opened_financial_year) unless financial_year&.opened?
+    errors.add(:printed_on, :not_opened_financial_year) unless financial_year&.opened? || financial_year&.closure_in_preparation?
```

Comment:
Can you wrap this in a `if(closure_preparation?)/else` instead of `unless ... closure_preparation?` + `if closure_preparation?` ?